### ISSUE TITLE
[Enhancement] Support Google Camera's formats (Photo Sphere, Moving Picture, ...)

### ISSUE THREAD

**Issue by GGORG0** [IacobIonut01/Gallery#395]
**Is your feature request related to a problem? Please describe.**
I'm using GrapheneOS on a Pixel 6 with the Google/Pixel Camera app. I still have to use Google Photos because it's so tightly integrated with GCam.

**Describe the solution you'd like**
Being able to view special media types, such as:
- Top Shot/Moving Picture (similar to Live Photos from iOS) - with a toggle similar to GPhotos to disable looping
- Panorama/Photo Sphere
- Icons for Portrait, Night Sight, Action Pan, Long Exposure, etc (signified by the last part of the filename, such as `PXL_20240416_182652344.PORTRAIT.jpg` )
- Timeline editors for Time Lapse and Slow Motion videos
- Anything else Google Photos does but I missed??

I've also noticed that when opening a photo right after taking it (in GCam/GPhotos) displays a "Processing..." popup in the lower left corner, and the quality of the photo improves drastically after 1-2 seconds. Is that performed by camera or photos?

**Describe alternatives you've considered**
Using Google Photos (without Google account or internet access) :/

**Comment by lantw44** [comment]
Some icons aren't linked to file names. For example:

- Night Sight photos have `NIGHT` in filenames if the Night Sight mode is explicitly selected, but they don't have special filenames if the mode is automatically enabled from the default Photo mode. 
- Face Unblur photos have no special filenames. It is automatically enabled and cannot be manually enabled.

It looks like motion photos aren't linked to file names, either. Recently I got a motion photo from a Google Photos share. Google Photos could play the video part online, but it couldn't after it was downloaded to a Pictures subfolder. The downloaded file has `MP` in the filename, and [GoMoPho](https://github.com/cliveontoast/GoMoPho) could extract the video part. Maybe Google Photos relies on some metadata instead of filenames to determine the type.

**Comment by IacobIonut01** [comment]
The file identification is mostly in the XMP metadata, I will have to extract that to identify.

**Comment by IacobIonut01** [comment]
Work in Progress

![Image](https://github.com/user-attachments/assets/e592574c-b7af-4b6c-8f00-7d6b18ebe917)

**Comment by IacobIonut01** [comment]
This is partially done in the latest release

**Comment by lantw44** [comment]
I downloaded Gallery-3.2.0-320064-arm64-v8a.apk, but the moon symbol looks wrong. A lot of photos which are not taken with night sight mode have moon symbols, and some night sight photos have no symbol.

**Comment by IacobIonut01** [comment]
> I downloaded Gallery-3.2.0-320064-arm64-v8a.apk, but the moon symbol looks wrong. A lot of photos which are not taken with night sight mode have moon symbols, and some night sight photos have no symbol.

Yes I know, that's why I've said partially implemented. The logic is not yet solid and since there's no documentation whatsoever for accurately knowing which photo is which kind, I'm currently trying to guess using some of the media metadata

**Comment by mon-jai** [comment]
It would be great if we could support viewing and extracting Top Shots (burst shots/motion photos). In that format, Google Camera attaches an MP4 file at the end of the output JPG file, which could later be used to extract alternative moments within the captured timeframe.

There is an implementation here. Maybe we could reuse it?  
https://github.com/googleinterns/motion-photos-filereader

**Comment by IacobIonut01** [comment]
> It would be great if we could support viewing and extracting Top Shots (burst shots/motion photos). In that format, Google Camera attaches an MP4 file at the end of the output JPG file, which could later be used to extract alternative moments within the captured timeframe.
> 
> There is an implementation here. Maybe we could reuse it?  
> https://github.com/googleinterns/motion-photos-filereader

Nice. Didn't know about it

**Comment by mon-jai** [comment]
Also, some context here: https://www.reddit.com/r/GooglePixel/comments/1anayjc/

Kindly tell me if you need any demo files. I own a Pixel device and have a lot of them.

**Comment by IacobIonut01** [comment]
> Also, some context here: https://www.reddit.com/r/GooglePixel/comments/1anayjc/
> 
> Kindly tell me if you need any demo files. I own a Pixel device and have a lot of them.

I'm also rocking a Pixel 6, so I've got that covered, but thanks!
